lib/traject/dor_config.rb

```diff
@@ -79,6 +79,19 @@
 # these are for single value facet display (in lieu of date slider (pub_year_tisim) )
 to_field 'pub_year_no_approx_isi', stanford_mods(:pub_year_int, true)
 to_field 'pub_year_w_approx_isi', stanford_mods(:pub_year_int, false)
+
+to_field 'date_display' do |resource, accumulator, _context|
+  imprint = ModsDisplay::Imprint.new(nil, ModsDisplay::Configuration::Imprint.new, nil)
+
+  Array(resource.smods_rec.origin_info).map do |value|
+    dates = imprint.dates(value)
+    accumulator.concat(dates.map(&:values).flatten)
+
+    part = imprint.send(:parts_element, value)
```

Comment:
is this calling a private method in the Imprint class? should we rework mods_display to make that a public method?

Reply:
Yes, we should. It's going to be an unfortunate amount of work, though, to refactor that code and reconcile it with stanford-mods (which seems to have a copy/pasted version of the code, which was later tweaked).